code/graph_algorithms/src/matrix_transformation/matrix_transformation.swift

```diff
@@ -0,0 +1,62 @@
+// Part of Cosmos by OpenGenus Foundation
+
+import Foundation
+
+class MatrixTransformation {
+    func rotate<T>(_ matrix: inout [[T]]) {
+        guard matrix.count > 0 && matrix.count == matrix[0].count else {
+            return
+        }
+
+        // 1 2 3      7 8 9      7 4 1
+        // 4 5 6  =>  4 5 6  =>  8 5 2
+        // 7 8 9      1 2 3      9 6 3
+        matrix.reverse()
+        symmetric(&matrix)
+    }
+
+    func reverse_rotate<T>(_ matrix: inout [[T]]) {
+        guard matrix.count > 0 && matrix.count == matrix[0].count else {
+            return
+        }
+
+        // 1 2 3      3 2 1      3 6 9
+        // 4 5 6  =>  6 5 4  =>  2 5 8
+        // 7 8 9      9 8 7      1 4 7
+        for i in 0..<matrix.count {
+            matrix[i].reverse()
+        }
+        symmetric(&matrix)
+    }
+
+    func symmetric<T>(_ matrix: inout [[T]]) {
```

Comment:
Maybe `transpose` would be better? The only problem is that the transpose is defined for any matrix and not only square ones.

Maybe `reflect` is more proper, as it is the reflection of the matrix over its main diagonal.

Reply:
I think transpose is better. Maybe the problem would be fixed by add 'square' as prefix of file/folder name.

Reply:
Because It will be incremented complexity of code and time (maybe) if I implement for any matrix.